### ISSUE TITLE
[Suggestion] Shipyard GUI Feedback

### ISSUE THREAD

**Issue by ProPeach** [tsunamayo/Starship-EVO#2374]
A couple of quick suggestions on what I think could help the user experience of the Shipyard block

First is selecting a direction for Forced Entry - The current blue arrow could be a little confusing as it could either point towards the direction that the ship enters, or show which way the ship will face when it enters the yard. I think that an Entry Plane rather than an Entry Direction would help with this.

![image](https://user-images.githubusercontent.com/27857711/84217480-2d630f00-aac4-11ea-9afc-39f10396eaab.png)
On the left is a shipyard with Force Enter Direction set to False so a ship can enter in any direction, and on the right is a shipyard with that setting set to True so a ship can only enter via the green plane.

**Comment by Garrett-C** [comment]
I have 2 points to add.

Firstly I think it may be worth looking into allowing the player to set one enter/exit plane or setting one enter plane and one exit plane, essentially creating a system where the ship can pass through.  For example if you compare a dry dock to a hangar. A ship could feasibly fly through a dry dock but can only enter and exit from one point on a standard hangar. The reason behind this distinction would be to help with AI behaviour. If a large ship docks in a dry dock and leaves does it try to turn around or does it just reverse out? If you could almost tell the AI how to enter and exit the dock it would probably be pretty good.

Secondly I agree with peach that the ability to hold the gravity block and see the gravity fields is more intuitive and useful than using the link tool. That said using the link tool sounds more handy when it comes to survival. Perhaps going with both would be ideal. Holding the block would allow you to see all the gravity regions and using the link tool would only allow you to see one. This same functionality could be applied to the shipyard area.

**Comment by tsunamayo** [comment]
Yes I agree with all.
For the view gizmo actually I wanted to add a button inside the shipyard core to display the gizmo. Then you can toggle it off. I might want to do that with other block.
The think with the link tool is I will extend that gizmo view to any other logical block, typically turret, computer ect. So for example you select the yoke with the link, it displays the arrows. So it will hopefully make a bit of sense even if there is no link for the shipyard (for now, as after I add the "input / event" gate you will link your sihpyard to other stuff like door, elevator, lights).
Thanks

**Comment by JRL101** [comment]
Is it also possible to have the shipyard module output signals? 
so we can do things like animations when something is being repaired etc?

**Comment by Lsutehall** [comment]
Having had a play with shipyards and gravity yesterday, I found it's but tricky and slightly annoying to figure out which of the coordinate boxes moved the field in which direction. I had to make a change, go observe it, and then go back to the core to change it again.

How would you feel about a "field block" - place and stretch like a block (but can clip through geometry to allow placement for gravity) and then link the relevant core to it via the link tool.

That would yield a visually-placeable component that uses the same mechanics as building, and such a block could be used for other "area effect" uses, depending on the core it's linked too - shipyard, gravity, force field, life support etc.

**Comment by tsunamayo** [comment]
@Lsutehall yes that was one of the options considered, but it is a bit clunky to explain to new players. I think I will go for a solution were you can select and drag handle on the side of each face of the box.
@JRL101 yes definitely, but I will do it in a generate way, each system would be able to ouput a set of events.
